core/src/main/java/org/apache/calcite/avatica/util/AbstractCursor.java

```diff
@@ -1220,7 +1227,7 @@ static class TimestampAccessor extends ObjectAccessor {
       }
       final long unix =
           DateTimeUtils.sqlTimestampToUnixTimestamp(timestamp, localCalendar);
-      return timestampAsString(unix, null);
+      return timestampAsString(unix, null, this.precision);
```

Comment:
For `cast(TIME '12:42:25.34' as TIME(2))` case,  whether to consider 
 `DateTimeUtils.unixTimeToSqlTime`  corresponding changes